tools/gdb/lists.py

```diff
@@ -0,0 +1,244 @@
+        c = n
+        nb += 1
+        if c == head:
+            gdb.write("list is consistent: {} node(s)\n".format(nb))
+            return
+
+
+def sq_for_every(sq, entry):
+    if sq.type == sq_queue.get_type().pointer():
+        sq = sq.dereference()
+    elif sq.type != sq_queue.get_type():
+        gdb.write("Must be struct sq_queue not {}".format(sq.type))
+        return
+
+    if sq["head"] == 0:
+        gdb.write(
+            "sq_for_every: Uninitialized queue '{}' treated as empty\n".format(
+                sq.address
+            )
+        )
+
+        return
+
+    entry = sq["head"].dereference()
+
+    while entry.address:
+        yield entry.address
+        entry = entry["flink"].dereference()
+
+
+def sq_is_empty(sq):
+    if sq.type == sq_queue.get_type().pointer():
+        sq = sq.dereference()
+    elif sq.type != sq_queue.get_type():
+        return False
+
+    if sq["head"] == 0:
+        return True
+    else:
+        return False
+
+
+def sq_check(sq):
+    nb = 0
+    if sq.type == sq_queue.get_type().pointer():
+        sq = sq.dereference()
+    elif sq.type != sq_queue.get_type():
+        gdb.write("Must be struct sq_queue not {}".format(sq.type))
+        return
+
+    if sq["head"] == 0:
+        gdb.write("sq_queue head is empty {}\n".format(sq.address))
+        return
+
+    entry = sq["head"].dereference()
+    try:
+        while entry.address:
+            nb += 1
+            entry = entry["flink"].dereference()
+    except gdb.MemoryError:
+        gdb.write("entry address is unaccessible {}\n".format(entry.address))
+        return
+
+    gdb.write("sq_queue is consistent: {} node(s)\n".format(nb))
+
+
+def dq_for_every(dq, entry):
+    if dq.type == dq_queue.get_type().pointer():
+        dq = dq.dereference()
+    elif dq.type != dq_queue.get_type():
+        gdb.write("Must be struct dq_queue not {}".format(dq.type))
+        return
+
+    if dq["head"] == 0:
+        gdb.write(
+            "dq_for_every: Uninitialized queue '{}' treated as empty\n".format(
+                dq.address
+            )
+        )
+
+        return
+    entry = dq["head"].dereference()
+    while entry.address:
+        yield entry.address
+        entry = entry["flink"].dereference()
+
+
+def dq_check(dq):
+    nb = 0
+    if dq.type == dq_queue.get_type().pointer():
+        dq = dq.dereference()
+    elif dq.type != dq_queue.get_type():
+        gdb.write("Must be struct dq_queue not {}".format(dq.type))
+        return
+
+    if dq["head"] == 0:
+        gdb.write("dq_queue head is empty {}\n".format(dq.address))
+        return
+    entry = dq["head"].dereference()
+    try:
+        while entry.address:
+            nb += 1
+            entry = entry["flink"].dereference()
+    except gdb.MemoryError:
+        gdb.write("entry address is unaccessible {}\n".format(entry.address))
+        return
+
+    gdb.write("dq_queue is consistent: {} node(s)\n".format(nb))
+
+
+class NxListChk(gdb.Command):
```

Comment:
Please include a single comment line before each function to describe its main purpose. It is important to have some hint about each function, imagine that someone else will need to do maintenance in this code in the future. Just a "# Search the python dir for all .py files, and source each" already helps a lot!

Maybe a single comment block at top of the file (before the functions will be enough). I don't know yet how GDB is works at that level, but I think this code need to transverse about single and double linked list (queue). Any explanation at the beginning is already encouraging for newcomers.

Reply:
Ok, I'll add some notes